src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/AttachCaseCallbackService.java

```diff
@@ -162,143 +186,203 @@ public AttachCaseCallbackService(CcdApi ccdApi, PaymentsProcessor paymentsProces
+            callBackEvent.exceptionRecordId
+        );
+
+        CaseDetails theCase = ccdApi.getCase(targetCaseCcdRef, callBackEvent.exceptionRecordJurisdiction);
+        List<Map<String, Object>> targetCaseDocuments = getScannedDocuments(theCase);
 
         //This is done so exception record does not change state if there is a document error
         checkForDuplicatesOrElse(
-            exceptionRecordDocuments,
+            callBackEvent.exceptionRecordDocuments,
             targetCaseDocuments,
             ids -> throwDuplicateError(targetCaseCcdRef, ids)
         );
 
         List<Map<String, Object>> newCaseDocuments = attachExceptionRecordReference(
-            exceptionRecordDocuments,
-            exceptionRecordId
+            callBackEvent.exceptionRecordDocuments,
+            callBackEvent.exceptionRecordId
         );
 
-        StartEventResponse event = ccdApi.startAttachScannedDocs(theCase, idamToken, userId);
+        StartEventResponse ccdStartEvent =
+            ccdApi.startAttachScannedDocs(theCase, callBackEvent.idamToken, callBackEvent.userId);
 
         Map<String, Object> newCaseData = buildCaseData(newCaseDocuments, targetCaseDocuments);
 
         ccdApi.attachExceptionRecord(
             theCase,
-            idamToken,
-            userId,
+            callBackEvent.idamToken,
+            callBackEvent.userId,
             newCaseData,
-            createEventSummary(theCase, exceptionRecordId, newCaseDocuments),
-            event
+            createEventSummary(theCase, callBackEvent.exceptionRecordId, newCaseDocuments),
+            ccdStartEvent
         );
 
         log.info(
             "Successfully attach Exception Record with ID {} to case with id {}",
             theCase.getId(),
-            exceptionRecord.getId()
+            exceptionRecordDetails.getId()
         );
 
-        paymentsProcessor.updatePayments(exceptionRecord, theCase.getId());
+        paymentsProcessor.updatePayments(exceptionRecordDetails, theCase.getId());
+    }
 
+    private void updateSupplementaryEvidenceWithOcr(
+        AttachToCaseEventData callBackEvent,
+        String targetCaseCcdRef,
+        CaseDetails exceptionRecordDetails,
+        boolean ignoreWarnings
+    ) {
+        ServiceConfigItem serviceConfigItem = getServiceConfig(exceptionRecordDetails);
+        Validation<Seq<String>, ExceptionRecord> exceptionRecordValidation =
+            exceptionRecordValidator.getValidation(exceptionRecordDetails);
+        if (exceptionRecordValidation.isValid()) {
+            ccdCaseUpdater.updateCase(
+                exceptionRecordValidation.get(),
+                serviceConfigItem,
+                ignoreWarnings,
+                callBackEvent.idamToken,
+                callBackEvent.userId,
+                targetCaseCcdRef
+            );
+        } else {
+            throw new CallbackException(exceptionRecordValidation.getError().mkString());
```

Comment:
This is a validation error - it needs to be returned by the caseworker within a 200 response, inside `errors` array. This exception will, in turn, result in a 500 response, which the caseworker will see as a generic error. To make things slightly worse, we'll log all these errors, as they make the exception message.